test/EventStore.Client.Streams.Tests/Subscriptions/subscribe_to_stream.cs

```diff
@@ -230,4 +230,43 @@ async Task Subscribe() {
 			}
 		}
 	}
+
+    [Fact]
+    [Trait("Category", "Special cases")]
+    [Description("This is normal behavior for EventStoreDB")]
+    public async Task should_throw_exception_when_subscribing_to_deleted_stream_category() {
+	    var category   = Guid.NewGuid().ToString("N");
+	    var streamName = category + "-123";
+
+	    var seedEvents = Fixture.CreateTestEvents(type: $"{category}-{Fixture.GetStreamName()}").ToArray();
+	    await Fixture.Streams.AppendToStreamAsync(streamName, StreamState.NoStream, seedEvents);
+	    await Fixture.Streams.DeleteAsync(streamName, StreamState.StreamExists);
+
+	    await using var subscription = Fixture.Streams.SubscribeToStream("$ce-" + category, FromStream.Start, resolveLinkTos: true);
+
+	    var ex = await Assert.ThrowsAsync<NullReferenceException>(
+		    async () => {
+			    await using var enumerator = subscription.Messages.GetAsyncEnumerator();
+
+			    Assert.True(await enumerator.MoveNextAsync());
+
+			    Assert.IsType<StreamMessage.SubscriptionConfirmation>(enumerator.Current);
+
+			    await Subscribe().WithTimeout();
+
+			    return;
+
+			    async Task Subscribe() {
+				    while (await enumerator.MoveNextAsync()) {
+					    if (enumerator.Current is not StreamMessage.Event(var resolvedEvent))
+						    continue;
+
+					    _ = resolvedEvent.Event.Data;
```

Comment:
I'm confused by this test because it should be about not throwing an NPE when collecting telemetry when dealing with an event from a deleted stream. I don't see the point of ascertaining how the server behaves when dealing with an event from a deleted stream.

Reply:
I removed the subscribe_to_stream test. Now, I only test to ensure that it doesn't throw an NPE when collecting telemetry